### PR TITLE
Add windows CI 2.x

### LINKED ISSUE #274: Add Windows Support
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/95, add Windows support.

- [ ] Passing CI on Windows
- [ ] Documentation

### PR DESCRIPTION
**Pull request by amitgalitz**
Signed-off-by: Amit Galitzky <amgalitz@amazon.com>

### Description
Adds windows CI for 2.x branch

1. Changed the usage of getPath() to first use .toURI() in some of the test cases. This is because just getting the path like we used to was returning a string like this:` \C:\Sample\sample.txt`. This isn't supported on Windows. Converting with `toUri()` works on all platforms.
### Issues Resolved
#274 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by amitgalitz** [comment]
3 failings tests in windows:
```
- org.opensearch.ad.ml.CheckpointDaoTests.testDeserializeRCFModelPreINIT
 - org.opensearch.ad.ml.CheckpointDaoTests.testDeserializeRCFModelPostINIT
 - org.opensearch.ad.ml.CheckpointDaoTests.testDeserializeTRCFModel
 ```

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -8,7 +8,35 @@ on:
...
+    name: Build and Test Anomaly Detection Plugin on Windows
+    runs-on: windows-latest
+
+    steps:
+      - name: Setup Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+
+      # ml-commons
```
Comment (ohltyler) [inline .github/workflows/CI.yml:25]: nit - can remove this comment

**Review by ohltyler — DISMISSED** [review]
One minor comment. LGTM!

File: .github/workflows/CI.yml (hunk 1 above)
Comment (amitgalitz) [reply .github/workflows/CI.yml]: fixed

**Review by ohltyler — APPROVED** [review]

**Comment by ohltyler** [comment]
Btw, do you know if we want to support for 3.0 as well? Wondering if it's worth front-porting there, and if it's not ready, could just add the changes, but have them commented out in the workflow file for now until that's possible.

**Comment by amitgalitz** [comment]
> Btw, do you know if we want to support for 3.0 as well? Wondering if it's worth front-porting there, and if it's not ready, could just add the changes, but have them commented out in the workflow file for now until that's possible.

Actually now that I check 3.0 has just been fixed. I can either forward this or close this. merge into main and backport to 2.x

**Comment by ohltyler** [comment]
> > Btw, do you know if we want to support for 3.0 as well? Wondering if it's worth front-porting there, and if it's not ready, could just add the changes, but have them commented out in the workflow file for now until that's possible.
> 
> Actually now that I check 3.0 has just been fixed. I can either forward this or close this. merge into main and backport to 2.x

Front porting should be fine.

**Comment by amitgalitz** [comment]
> > > Btw, do you know if we want to support for 3.0 as well? Wondering if it's worth front-porting there, and if it's not ready, could just add the changes, but have them commented out in the workflow file for now until that's possible.
> > 
> > 
> > Actually now that I check 3.0 has just been fixed. I can either forward this or close this. merge into main and backport to 2.x
> 
> Front porting should be fine.

missed this haha, I guess we will merge the other one already.

**Pull request closed without merging**